pkg/handlers/v1/attribute.go

```diff
@@ -21,23 +22,24 @@ type AttributeHandler struct {
 func (h *AttributeHandler) Handle(ctx context.Context, assetVulns domain.NexposeAssetVulnerabilities) error {
 	logger := h.LogFn(ctx)
 
-	attributedAssetVulns, err := h.AssetAttributor.Attribute(ctx, assetVulns)
-	if err != nil {
-		switch err.(type) {
+	attributedAssetVulns, attributionErr := h.AssetAttributor.Attribute(ctx, assetVulns)
+	if attributionErr != nil {
+		switch attributionErr.(type) {
 		case domain.AssetNotFoundError:
-			logger.Error(logs.AssetNotFoundError{Reason: err.Error()})
-			return err
+			logger.Error(logs.AssetNotFoundError{Reason: attributionErr.Error()})
+			err := h.AttributionFailureHandler.HandleAttributionFailure(ctx, attributedAssetVulns)
```

Comment:
Question: So we can only handle AssetNotFound errors? Am I reading this right? I'd expect us to handle attribution failures in the `default` case here, not the AssetNotFoundError case.

Reply:
Should we handle all cases?

Reply:
In this codebase, there is no such concept as "asset not found".  This microservice is processing messages from Nexpose.  There is no "fetch asset" occurring in this microservice, hence no such thing as "asset not found".

Reply:
what would you have me do?

Reply:
> you

I think you don't need to add any "asset not found" handlers, but also seek @LorenBoren's take.

Reply:
To be fair, I think there is absolutely an "asset not found" concept in this service, but it means something different in the attribution context than it means in the hydration context.

I would expect all non-recoverable errors to initiate attribution failure handling, because those are the only ones that can be certain will require alternative processing. In this case, the non-recoverable errors are `AssetNotFoundError` and `AssetInventoryMultipleAssetsFoundError`. I propose invoking the `AttributionFailureHandler` in just these cases. 

What does the rest of the team think?

Reply:
I'm good with that - just the AssetNotFoundError and AssetInventoryMultipleAssetsFoundError cases